backend/jobs/log-alerts/log-alerts.go

```diff
@@ -0,0 +1,194 @@
+						processLogAlerts(ctx, DB, TDB, MailClient, alerts, rh, redis, ccClient)
+					}()
+				}
+			}
+		}()
+	}
+}
+
+func getLogAlerts(ctx context.Context, DB *gorm.DB) []*model.LogAlert {
+	var alerts []*model.LogAlert
+	if err := DB.Model(&model.LogAlert{}).Where("disabled = ?", false).Find(&alerts).Error; err != nil {
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
+			log.WithContext(ctx).Error("Error querying for log alerts")
+		}
+	}
+
+	return alerts
+}
+
+func processLogAlerts(ctx context.Context, DB *gorm.DB, TDB timeseries.DB, MailClient *sendgrid.Client, logAlerts []*model.LogAlert, rh *resthooks.Resthook, redis *redis.Client, ccClient *clickhouse.Client) {
+	log.WithContext(ctx).Info("Number of Log Alerts to Process: ", len(logAlerts))
+	for _, alert := range logAlerts {
+		lastTs, err := redis.GetLastLogTimestamp(ctx, alert.ProjectID)
+		if err != nil {
+			log.WithContext(ctx).Error("error retrieving last log timestamp", err)
+			continue
+		}
+		// If there's no log timestamp set, assume time.Now()
+		if lastTs.IsZero() {
+			lastTs = time.Now()
+		}
+		end := lastTs.Add(-time.Minute)
+		start := end.Add(-time.Minute)
+
+		count64, err := ccClient.ReadLogsTotalCount(ctx, alert.ProjectID, modelInputs.LogsParamsInput{Query: alert.Query, DateRange: &modelInputs.DateRangeRequiredInput{
+			StartDate: start,
+			EndDate:   end,
+		}})
+		if err != nil {
+			log.WithContext(ctx).Error("error querying clickhouse for log count", err)
+			continue
+		}
+		count := int(count64)
+
+		alertCondition := count >= alert.CountThreshold
+		if alert.BelowThreshold {
+			alertCondition = count <= alert.CountThreshold
+		}
+
+		if alertCondition {
+			var project model.Project
+			if err := DB.Model(&model.Project{}).Where("id = ?", alert.ProjectID).First(&project).Error; err != nil {
+				log.WithContext(ctx).Error("error querying project for processMetricMonitor", err)
+				continue
+			}
+			var workspace model.Workspace
+			if err := DB.Where(&model.Workspace{Model: model.Model{ID: project.WorkspaceID}}).First(&workspace).Error; err != nil {
+				log.WithContext(ctx).Error("error querying workspace for processMetricMonitor", err)
+				continue
+			}
+
+			aboveStr := "above"
+			if alert.BelowThreshold {
+				aboveStr = "below"
+			}
+
+			hookPayload := zapier.HookPayload{
+				MetricValue:     pointy.Float64(float64(count)),
+				MetricThreshold: pointy.Float64(float64(alert.CountThreshold)),
+			}
+			if err := rh.Notify(project.ID, fmt.Sprintf("LogAlert_%d", alert.ID), hookPayload); err != nil {
+				log.WithContext(ctx).Error("error notifying zapier", err)
+			}
+
+			queryStr := ""
+			if alert.Query != "" {
+				queryStr = fmt.Sprintf(`for query *%s* `, alert.Query)
+			}
+			message := fmt.Sprintf(
+				"🚨 *%s* fired!\nLog count %sis currently %s the threshold.\n"+
+					"_Count_: %d | _Threshold_: %d",
+				*alert.Name,
+				queryStr,
+				aboveStr,
+				count,
+				alert.CountThreshold,
+			)
+
+			fmt.Println(message)
```

Comment:
should we logrus log instead with additional context?

Reply:
yep will do! copied this over from metric monitors